packages/phone-number-privacy/combiner/src/salt-generation/get-threshold-salt.ts

```diff
@@ -0,0 +1,68 @@
+import { Request, Response } from 'firebase-functions'
+import fetch from 'node-fetch'
+import { BLSCryptographyClient, ServicePartialSignature } from '../bls/bls-cryptography-client'
+import { ErrorMessages, respondWithError } from '../common/error-utils'
+import { authenticateUser } from '../common/identity'
+import {
+  hasValidAccountParam,
+  hasValidQueryPhoneNumberParam,
+  isBodyReasonablySized,
+  phoneNumberHashIsValidIfExists,
+} from '../common/input-validation'
+import config from '../config'
+
+const PARTIAL_SIGN_MESSAGE_ENDPOINT = '/getBlindedSalt'
+export interface SignMessageResponse {
+  success: boolean
+  signature: string
+}
+
+export async function handleGetDistributedBlindedMessageForSalt(
+  request: Request,
+  response: Response
+) {
+  if (!isValidGetSignatureInput(request.body)) {
+    respondWithError(response, 400, ErrorMessages.INVALID_INPUT)
+    return
+  }
+  if (!authenticateUser(request)) {
+    respondWithError(response, 401, ErrorMessages.UNAUTHENTICATED_USER)
+    return
+  }
+  const signatures: ServicePartialSignature[] = []
+  for (const service of config.pgpnpServices) {
```

Comment:
We should make each API call in parallel to reduce the overall latency. @tkporter has a good example here:
https://github.com/celo-org/celo-oracle/blob/2678f7046100f337dd46bdaac8ba8b969011a8bc/src/data_aggregator.ts#L269

https://github.com/celo-org/celo-oracle/blob/2678f7046100f337dd46bdaac8ba8b969011a8bc/src/utils.ts#L352

Reply:
great callout 👍